### PR TITLE
Fixture attributes implemented

### PR DESCRIPTION
**Pull request by olegpidsadnyi**

### CODE REVIEW

**Comment by olegpidsadnyi** [comment]
Doesn't work with post generation :(

**Comment by olegpidsadnyi** [comment]
Should work now

File: pytest_factoryboy/fixture.py [hunk 1]
```diff
@@ -138,23 +144,35 @@ def attributes(cls, create=False, extra=None):
...
+            if isinstance(declaration, factory.RelatedFactory):
+                related.append(attr)
+            else:
+                postgen.append(attr)
 
     result = Factory(**data)
     if postgen_declarations:
-        request._fixturedef.cached_result = (result, 0, None)
-        request._fixturedefs[request.fixturename] = request._fixturedef
+        factoryboy_request = request.getfuncargvalue("factoryboy_request")
```
Comment (bubenkoff) [inline pytest_factoryboy/fixture.py]: extract to separate function, the outer function became unreadable and too complex

File: pytest_factoryboy/fixture.py [hunk 2]
```diff
@@ -138,23 +144,35 @@ def attributes(cls, create=False, extra=None):
...
 
-        for attr, declaration in postgen_declarations.items():
-            if isinstance(declaration, factory.RelatedFactory):
-                request.getfuncargvalue(prefix + attr)
-            else:
-                extra = postgen_declaration_args[attr].extra
-                declaration.function(result, True, request.getfuncargvalue(prefix + attr), **extra)
+        def deferred():
```
Comment (bubenkoff) [inline pytest_factoryboy/fixture.py]: can be separate function as well

File: pytest_factoryboy/fixture.py (hunk 1 above)
Comment (olegpidsadnyi) [inline pytest_factoryboy/fixture.py]: it uses local vars in every line. separate function becomes too ugly

File: pytest_factoryboy/plugin.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    def __init__(self):
+        self.deferred = []
+        self.is_evaluated = False
+
+    def defer(self, function):
+        """Defer post-generation declaration execution until the end of the test setup.
+
+        :param function: Function to be executed.
+        """
+        if self.is_evaluated:
```
Comment (bubenkoff) [inline pytest_factoryboy/plugin.py]: conflicts with the docstring, kind of

File: pytest_factoryboy/plugin.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    """PyTest FactoryBoy request fixture."""
+    return Request()
+
+
+@pytest.mark.tryfirst
+def pytest_runtest_call(item):
+    """Before the test item is called."""
+    try:
+        request = item._request
+    except AttributeError:
```
Comment (bubenkoff) [inline pytest_factoryboy/plugin.py]: comment why can it be

**Pull request merged**